src/languages/java.rs

```diff
@@ -18,42 +18,78 @@ impl JavaProcessor {
         Ok(Self { parser })
     }
 
-    fn find_all_methods<'a>(&self, root_node: Node<'a>) -> Vec<Node<'a>> {
-        let mut methods = Vec::new();
-        let mut cursor = root_node.walk();
-
-        Self::traverse_for_methods(&mut cursor, &mut methods);
-
-        methods
-    }
-
-    fn traverse_for_methods<'a>(cursor: &mut TreeCursor<'a>, methods: &mut Vec<Node<'a>>) {
-        if cursor.node().kind() == "method_declaration" {
-            methods.push(cursor.node());
-        }
-
+    /// Helper method to traverse all children of a node recursively.
+    /// This avoids code duplication in the main traversal method.
+    fn traverse_children<'a>(
+        &self,
+        node: Node<'a>,
+        content: &str,
+        stack: &mut Vec<(Node<'a>, String)>,
+        chunks: &mut Vec<CodeChunk>,
+    ) {
+        let mut cursor = node.walk();
         if cursor.goto_first_child() {
             loop {
-                Self::traverse_for_methods(cursor, methods);
+                self.collect_chunks_recursively(cursor.node(), content, stack, chunks);
                 if !cursor.goto_next_sibling() {
                     break;
                 }
             }
-            cursor.goto_parent();
         }
     }
 
-    fn find_enclosing_class_or_interface<'a>(&self, method_node: Node<'a>) -> Option<Node<'a>> {
-        let mut current = method_node;
-
-        while let Some(parent) = current.parent() {
-            if parent.kind() == "class_declaration" || parent.kind() == "interface_declaration" {
-                return Some(parent);
+    /// Recursively traverses the AST to collect code chunks.
+    /// The stack maintains the current nesting context of classes/interfaces
+    /// to provide proper context for method declarations.
+    fn collect_chunks_recursively<'a>(
+        &self,
+        node: Node<'a>,
+        content: &str,
+        stack: &mut Vec<(Node<'a>, String)>, // Stack of (container_node, container_name) for nesting context
+        chunks: &mut Vec<CodeChunk>,
+    ) {
+        match node.kind() {
+            "class_declaration" | "interface_declaration" => {
+                let container_name = self.get_container_name(node, content);
+                stack.push((node, container_name));
+
+                // Process children
+                self.traverse_children(node, content, stack, chunks);
+
+                stack.pop();
+            }
+            "method_declaration" => {
+                // Extract method name, fallback to a default if parsing fails
+                let method_name = self.get_method_name(node, content)
+                    .unwrap_or_else(|| format!("method_at_line_{}", node.start_position().row + 1));
```

Comment:
Ignore lambdas (anonymous methods). If they are represented in TreeSitter's AST as method nodes without a name (verify if this is the case)

Anonymous classes are fine though, we still want methods of anonymous classes indexed